### DISCUSSION TITLE
Does successful NRF connection status imply HW is wired correctly

### DISCUSSION THREAD

**Discussion by pkese** [tbnobody/OpenDTU#1047] [Q&A]
I'm trying to diagnose why doesn't my NRF24L01+ module connect to solar inverter.

Does the successful connection status reported by OpenDTU imply that hardware is wired correctly (I have also added 47uF and 100nF capacitors between ESP32 and NRF - right next to NFR pins)?
![image](https://github.com/tbnobody/OpenDTU/assets/278492/2261e063-a240-4203-9a30-b5df376f901b)

Is this what the serial number of Hoymiles HM800 should look like: `114193058884`?

**Comment by pkese** [comment]
Is it a problem that I have the nRF24L01+ transciever with LNA and external antenna?

Somebody here is saying that only the model with built-in antenna works for him... https://github.com/tbnobody/OpenDTU/issues/193#issuecomment-1276148664

**Comment by peff74** [comment]
Hi,
the SN seams to be ok.
What does the console say?

Regarding the RF24 modules... These have also cost me my nerves.
There are some that work, others are not even recognized by the system.
And still others pretend to work, but then start spinning in between.
In addition, they are very sensitive to disturbing impulses.

This video helps a lot to understand other problems with the NF24 modules
[NRF24 Frustration - Radio module doesn't work?](https://www.youtube.com/watch?v=Z7_Cy66Vnrc)

**Comment by pkese** [comment]
Thanks for suggestions. Here's serial diagnostics.
```
rst:0xc (SW_CPU_RESET),boot:0x13 (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:2
load:0x3fff0030,len:1184
load:0x40078000,len:13192
load:0x40080400,len:3028
entry 0x400805e4
E (625) esp_core_dump_flash: No core d▒f▒▒partition found!
E (625) esp_core_dump_flash: No core dump partition found!

Starting OpenDTU
Initialize FS... done
Reading configuration... done
Reading PinMapping... [   166][E][vfs_api.cpp:105] open(): /littlefs/pin_mapping.json does not exist, no permits for creation
using default config done
Initialize Network... done
Setting Hostname... Configuring WiFi STA using new credentials... done
Initialize NTP... done
Initialize SunPosition... done
Initialize MqTT... done
Initialize WebApi... done
Initialize Display... done
Initialize LEDs... done
Check for default DTU serial... done
Initialize Hoymiles interface... NRF: Connection successful
  Setting radio PA level...
  Setting DTU serial...
  Setting poll interval...
  Adding inveWiFi connected
rter: 114183078886 - Steha done
done
Switch to WiFi mode
Setting Hostname... done
WiFi got ip: 192.168.0.104
Network connected
Configuring WiFi STA using existing credentials... done
Configuring WiFi STA static IP... WiFi got ip: 192.168.0.9
Network connected
done
Fetch inverter: 114183078886
Request SystemConfigPara
TX RealTimeRunData Channel: 23 --> 15 83 07 88 86 80 14 86 96 80 0B 00 64 8D D3 02 00 00 00 00 00 00 00 00 2D AF 2A
RX Period End
All missing
Nothing received, resend whole request
TX RealTimeRunData Channel: 40 --> 15 83 07 88 86 80 14 86 96 80 0B 00 64 8D D3 02 00 00 00 00 00 00 00 00 2D AF 2A
RX Period End
All missing
Nothing received, resend whole request
TX RealTimeRunData Channel: 61 --> 15 83 07 88 86 80 14 86 96 80 0B 00 64 8D D3 02 00 00 00 00 00 00 00 00 2D AF 2A
RX Period End
All missing
Nothing received, resend whole request
TX RealTimeRunData Channel: 75 --> 15 83 07 88 86 80 14 86 96 80 0B 00 64 8D D3 02 00 00 00 00 00 00 00 00 2D AF 2A
RX Period End
All missing
Nothing received, resend whole request
TX RealTimeRunData Channel: 3 --> 15 83 07 88 86 80 14 86 96 80 0B 00 64 8D D3 02 00 00 00 00 00 00 00 00 2D AF 2A
RX Period End
All missing
Nothing received, resend count exeeded
TX AlarmData Channel: 23 --> 15 83 07 88 86 80 14 86 96 80 11 00 64 8D D3 02 00 00 00 00 00 00 00 00 F7 B4 F1
RX Period End
All missing
Nothing received, resend whole request
TX AlarmData Channel: 40 --> 15 83 07 88 86 80 14 86 96 80 11 00 64 8D D3 02 00 00 00 00 00 00 00 00 F7 B4 F1
RX Period End
All missing
Nothing received, resend whole request
TX AlarmData Channel: 61 --> 15 83 07 88 86 80 14 86 96 80 11 00 64 8D D3 02 00 00 00 00 00 00 00 00 F7 B4 F1
RX Period End
All missing
Nothing received, resend whole request
TX AlarmData Channel: 75 --> 15 83 07 88 86 80 14 86 96 80 11 00 64 8D D3 02 00 00 00 00 00 00 00 00 F7 B4 F1
RX Period End
All missing
Nothing received, resend whole request
TX AlarmData Channel: 3 --> 15 83 07 88 86 80 14 86 96 80 11 00 64 8D D3 02 00 00 00 00 00 00 00 00 F7 B4 F1
RX Period End
All missing
Nothing received, resend count exeeded
TX SystemConfigPara Channel: 23 --> 15 83 07 88 86 80 14 86 96 80 05 00 64 8D D3 02 00 00 00 00 00 00 00 00 E3 A0 E5
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 40 --> 15 83 07 88 86 80 14 86 96 80 05 00 64 8D D3 02 00 00 00 00 00 00 00 00 E3 A0 E5
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 61 --> 15 83 07 88 86 80 14 86 96 80 05 00 64 8D D3 02 00 00 00 00 00 00 00 00 E3 A0 E5
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 75 --> 15 83 07 88 86 80 14 86 96 80 05 00 64 8D D3 02 00 00 00 00 00 00 00 00 E3 A0 E5
RX Period End
All missing
Nothing received, resend whole request
TX SystemConfigPara Channel: 3 --> 15 83 07 88 86 80 14 86 96 80 05 00 64 8D D3 02 00 00 00 00 00 00 00 00 E3 A0 E5
RX Period End
All missing
Nothing received, resend count exeeded
```

**Comment by pkese** [comment]
My board is TTGO Koala which has a reasonably decent power supply with SY8089 with an onboard inductor. Apparently it is supposed to be able to source 2A of current, which I assume should be enough for nRF module.

![image](https://github.com/tbnobody/OpenDTU/assets/278492/1197ba0e-e856-4375-a0d0-436e040affaf)

(That bottom pin is normally connected and wires seem to be holding fine - I disconnected it just when tying to lay everything flat while taking a photo).

**Comment by pkese** [comment]
I'll try to order some other modules - ideally the ones with proper RF shielding.

**Comment by pkese** [comment]
Maybe the 'decent' switching power supply is exactly the problem with this setup.  
There's an interesting discussion explaining the problems with power supply ripple on these boards here https://hackaday.com/2016/05/31/fixing-the-terrible-range-of-your-cheap-nrf24l01-palna-module/#comment-3039168

**Comment by peff74** [comment]
I also had problems with a switching power supply because I wanted to use POE. It didn't work at all.
I usually use WT32-ETH01 with one of those cheap POE to 5V adapters.
Has worked in all scenarios so far.
Not here!
Now it runs with a simple ESP32 on a Samsung mobile phone charger, which works.
But I am always working on an implementation with POE.
